tests/base.py

```diff
@@ -0,0 +1,444 @@
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"date_updated"},
+                self.RESPECTS_START_DATE: True,
+            },
+
+            'tasks': {
+                self.AUTOMATIC_FIELDS: {
+                    'date_created',
+                    'date_updated',
+                    'id',
+                    'is_complete',
+                    'phone',
+                    'local_phone',
+                    'voicemail_url',
+                    'organization_id',
+                    '_type',
+                    'date',
+                    'voicemail_duration',
+                    'opportunity_value_currency',
+                    'contact_name',
+                    'body_preview',
+                    'phone_number_description',
+                    'text',
+                    'opportunity_note',
+                    'due_date',
+                    'updated_by',
+                    'opportunity_value',
+                    'email_id',
+                    'remote_phone',
+                    'created_by',
+                    'view',
+                    'object_type',
+                    'subject',
+                    'remote_phone_description',
+                    'contact_id',
+                    'emails',
+                    'lead_id',
+                    'recording_url',
+                    'remote_phone_formatted',
+                    'assigned_to',
+                    'is_dateless',
+                    'object_id',
+                    'opportunity_value_formatted',
+                    'updated_by_name',
+                    'phone_formatted',
+                    'created_by_name',
+                    'lead_name',
+                    'assigned_to_name',
+                    'opportunity_value_period',
+                },
+                self.PRIMARY_KEYS: {"id"},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"date_updated"},
+                self.RESPECTS_START_DATE: True,
+            },
+
+            'users': {
+                self.AUTOMATIC_FIELDS: {
+                    'date_updated',
+                    'id',
+                    'last_name',
+                    'date_created',
+                    'organizations',
+                    'email',
+                    'first_name',
+                    'image',
+                    'last_used_timezone',
+                },
+                self.PRIMARY_KEYS: {"id"},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"date_updated"},
+                self.RESPECTS_START_DATE: True,
+            }
+
+        }
+
+    def expected_automatic_fields(self):
+        auto_fields = {}
+        for k, v in self.expected_metadata().items():
+            auto_fields[k] = v.get(self.AUTOMATIC_FIELDS, set())
+
+        return auto_fields
+
+
+    @classmethod
+    def setUpClass(cls):
```

Comment:
This can be simplified to

```
if not all(required_env_var in os.environ() for required_env_var in ['TAP_CLOSEIO_API_KEY']):
    # raise 
```

Reply:
I agree with you, but the `not all()` doesn't allow you to get the missing variables. Granted, it's one variable here, but for consistency, I think we should leave this as is